tests/TestCase/Network/Email/EmailTest.php

```diff
@@ -2592,6 +2596,74 @@ public function testZeroOnlyLinesNotBeingEmptied()
     }
 
     /**
+     * testJsonSerialize()
+     *
+     * @return void
+     */
+    public function testJsonSerialize()
+    {
+        $xmlstr = <<<XML
+<?xml version='1.0' standalone='yes'?>
+<framework>
+    <name>CakePHP</name>
+    <url>http://cakephp.org</url>
+</framework>
+XML;
+
+        $this->CakeEmail->reset()
+            ->to(['cakephp@cakephp.org' => 'CakePHP'])
+            ->from('noreply@cakephp.org')
+            ->replyTo('cakephp@cakephp.org')
+            ->cc(['mark@cakephp.org', 'juan@cakephp.org' => 'Juan Basso'])
+            ->bcc('phpnut@cakephp.org')
+            ->subject('Test Serialize')
+            ->template('default', 'test')
+            ->messageId('<uuid@server.com>')
+            ->domain('foo.bar')
+            ->viewVars([
+                'users' => TableRegistry::get('Users')->get(1, ['fields' => ['id', 'username']]),
+                'file' => fopen(TEST_APP . 'config' . DS . 'empty.ini', 'r'),
```

Comment:
Look closer ;) That's what am doing since there is no way to detect binary.

The ugliness comes from the fact that i had to base64 decode whatever was encoded or suggest leaving that to the user?